nodeJS/express/express_forms_validation.md

```diff
@@ -0,0 +1,565 @@
+- `method`: Defines the HTTP method to use (`POST` or `GET`).
+
+`POST` is generally more secure because it keeps sensitive information out of the URL, which means they won't show up in server logs, and is the standard choice for creating or updating data on the server side. `GET` is for forms that don't modify data, such as search forms, or when you want the form submission to be bookmarkable or shareable via URL.
+
+#### Form handling process
+
+The form's `action` will point to some endpoint on our server, which will allow our controller the request. The controller then communicates with the database to handle the data.
+
+We then generate a new or updated view with the controller's response and redirect the client. This is known as the [Post/Redirect/Get (PRG)](https://en.wikipedia.org/wiki/Post/Redirect/Get) design pattern which helps prevent duplicate `POST` requests.
+
+### Validation and sanitization
+
+Before the data from a form is sent off to our server, we must first do two things
+
+- *Validation* ensures user input meets the specified criteria (e.g. required fields, correct format).
+- *Sanitization* cleans user input to prevent malicious data from being processed by removing or encoding potentially malicious characters.
+
+The library we'll be using is called `express-validator`. While it simplifies both of these processes for us, it's important to understand the underlying concepts of these two operations.
+
+#### Installation
+
+We start as usual by installing the correct package in the *root* folder of our project.
+
+```bash
+npm install express-validator
+```
+
+To get started using the package, we first need to *require* it in our project
+
+```javascript
+const { body, validationResult } = require("express-validator");
+```
+
+`express-validator` comes with many functions for just about every form operation you could think of, but for this lesson, we'll be sticking with `body()` and `validationResult()`.
+
+### Understanding the body() function
+
+The `body()` function allows you to specify which fields in the request body should be validated and sanitized, as well as how to handle it.
+
+```javascript
+[
+  body("birthdate", "Must be a valid date.")
+          .optional({ values: "falsy" })
+          .isISO8601() // Enforce a YYYY-MM-DD format.
+];
+```
+
+This example marks `birthdate` field as optional, but still enforces the ISO8601 date format on inputs. This is because `{ values: "falsy" }` means values that aren't `undefined`, `null`, `false`, or `0` will still be validated.
+
+### Chaining validations
+
+You can also chain multiple validation methods, with unique error messages if the checks fail.
+
+```javascript
+[
+    body("name")
+    .trim()
+    .isLength({ min: 1 })
+    .withMessage("Name can not be empty.")
+    .isAlpha()
+    .withMessage("Name must only contain alphabet letters."),  
+];
```

Comment:
Indentation, and also suggestion to use `notEmpty()` instead of `isLength({ min: 1 })`, thought that part isn't a blocker.
```suggestion
[
  body("name")
    .trim()
    .notEmpty()
    .withMessage("Name can not be empty.")
    .isAlpha()
    .withMessage("Name must only contain alphabet letters."),  
];
```